src/libcore/slice/mod.rs

```diff
@@ -1596,9 +1596,10 @@ impl<T> [T] {
     /// less than or equal to any value at a position `j > index`. Additionally, this reordering is
     /// unstable (i.e. any number of equal elements may end up at position `index`), in-place
     /// (i.e. does not allocate), and `O(n)` worst-case. This function is also/ known as "kth
-    /// element" in other libraries. The function returns a triplet of the values: all elements less
+    /// element" in other libraries. It returns a triplet of the following values: all elements less
     /// than the one at the given index, the value at the given index, and all elements greater than
-    /// the one at the given index.
+    /// the one at the given index. Panics when `index >= len()`, meaning it always panics on empty
+    /// slices.
```

Comment:
Panics are usually documented with their own section. See for example Vec::remove.

Reply:
Done.